src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -74,17 +74,25 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
                 return;
             }
 
-            // RegisterWorkspaceHost is required to be called from UI thread so push the code
-            // to UI thread to run. 
-            await Task.Factory.SafeStartNew(() =>
+            // Create a connection to the host in the BG to avoid taking the hit of loading 
+            // service hub on the UI thread.
+            using (var connection = await client.TryCreateConnectionAsync(WellKnownRemoteHostServices.RemoteHostService, CancellationToken.None).ConfigureAwait(false))
             {
-                var projectTracker = vsWorkspace.GetProjectTrackerAndInitializeIfNecessary(Shell.ServiceProvider.GlobalProvider);
-
                 var host = new WorkspaceHost(vsWorkspace, client);
 
-                projectTracker.RegisterWorkspaceHost(host);
-                projectTracker.StartSendingEventsToWorkspaceHost(host);
-            }, CancellationToken.None, ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.TaskScheduler).ConfigureAwait(false);
+                // RegisterWorkspaceHost is required to be called from UI thread so push the code
+                // to UI thread to run. 
+                await Task.Factory.SafeStartNew(() =>
+                {
+                    host.SetPreferredConnection(connection);
+                    var projectTracker = vsWorkspace.GetProjectTrackerAndInitializeIfNecessary(Shell.ServiceProvider.GlobalProvider);
+
+                    projectTracker.RegisterWorkspaceHost(host);
+                    projectTracker.StartSendingEventsToWorkspaceHost(host);
+
+                    host.SetPreferredConnection(null);
```

Comment:
:bulb: This should be in a `finally` block

:question: Why are we not setting this to null before calling `RegisterWorkspaceHost`? It appears that the host is exposed to other code by that call, making it possible for unrelated code to use the preferred connection where it isn't expected to.